### PR TITLE
feat (#354): Add the ability to use annotationless style with generic

### PR DESCRIPTION
**Pull request by iocanel**
java applications. Add vertx examples.

### CODE REVIEW

**Comment by iocanel** [comment]
@jponge: This is what I have in mind. wdyt?

File: readme.md [hunk 1]
```diff
@@ -1067,6 +1079,46 @@ Also it will look for the same files under the kubernetes profile:
 - application-kubernetes.yml
 - application-kubernetes.yaml
 
+##### Vert.x & generic Java
+
+For generic java, if the @Dekorate annotion is present, then dekorate will look for confiugration uder:
+
+- application.properties
```
Comment (jponge) [inline readme.md]: Does it also work for `application.yaml`? (I suppose yes)

File: readme.md [hunk 2]
```diff
@@ -1067,6 +1079,46 @@ Also it will look for the same files under the kubernetes profile:
...
+or using yaml:
+
+    dekorate:
+      openshift:
+        labels:
+          - key: foo
+            value: bar
+  
+#### related examples
+ - [vertx on kubernetes example](examples/vertx-on-kubernetes-example)
```
Comment (jponge) [inline readme.md]: `Vert.x` rather than `vertx` :-)

File: readme.md (hunk 1 above)
Comment (iocanel) [reply readme.md]: Yes it does.

**Comment by jponge** [comment]
Very cool. 

Let me try the PR locally.

**Comment by jponge** [comment]
Works great! 👍 

Given:

```java
@Dekorate
public class MainVerticle extends AbstractVerticle {

  @Override
  public void start(Promise<Void> startPromise) {
    vertx.createHttpServer()
      .requestHandler(this::handleRequest)
      .listen(8080, ar -> {
        if (ar.succeeded()) {
          startPromise.complete();
        } else {
          startPromise.fail(ar.cause());
        }
      });
  }

  private void handleRequest(HttpServerRequest request) {
    request.response()
      .putHeader("Content-Type", "plain/text")
      .end("Yo");
  }

  public static void main(String[] args) {
    Vertx.vertx().deployVerticle(new MainVerticle());
  }
}
```

and:

```yaml
dekorate:
  kubernetes:
    serviceType: "LoadBalancer"
    ports:
      - name: "http"
        containerPort: 8080
```

I got:

```yaml
---
apiVersion: "v1"
kind: "List"
items:
- apiVersion: "v1"
  kind: "Service"
  metadata:
    labels:
      app: "dekorate-and-vertx"
      version: "0.0.0-SNAPSHOT"
      group: "jponge"
    name: "dekorate-and-vertx"
  spec:
    ports:
    - name: "http"
      port: 8080
      targetPort: 8080
    selector:
      app: "dekorate-and-vertx"
      version: "0.0.0-SNAPSHOT"
      group: "jponge"
    type: "LoadBalancer"
- apiVersion: "apps/v1"
  kind: "Deployment"
  metadata:
    labels:
      app: "dekorate-and-vertx"
      version: "0.0.0-SNAPSHOT"
      group: "jponge"
    name: "dekorate-and-vertx"
  spec:
    replicas: 1
    selector:
      matchLabels:
        app: "dekorate-and-vertx"
        version: "0.0.0-SNAPSHOT"
        group: "jponge"
    template:
      metadata:
        labels:
          app: "dekorate-and-vertx"
          version: "0.0.0-SNAPSHOT"
          group: "jponge"
      spec:
        containers:
        - env:
          - name: "KUBERNETES_NAMESPACE"
            valueFrom:
              fieldRef:
                fieldPath: "metadata.namespace"
          image: "jponge/dekorate-and-vertx:0.0.0-SNAPSHOT"
          imagePullPolicy: "IfNotPresent"
          name: "dekorate-and-vertx"
          ports:
          - containerPort: 8080
            name: "http"
            protocol: "TCP"
```

**Pull request merged**